File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        def backtrack(index, cur):
            res.append(cur[:])

            for i in range(index, len(nums)):
                cur.append(nums[i])
                backtrack(i + 1, cur)
                cur.pop()

        res = []
        backtrack(0, [])

        return res

File: 78-subsets/subsets.php
class Solution {

	/**
	* @param Integer[] $nums
	* @return Integer[][]
	*/
	function subsets($nums) {

		$ans = [];
        $backtrack=function($cur,$index)use($nums,&$ans,&$backtrack){
            $ans[]=$cur;
            for ($i = $index; $i < count($nums); $i++) {
                $cur[]=$nums[$i];
                $backtrack($cur,$i+1);
                array_pop($cur);
            }
        };
        
		$backtrack([],0);
        return $ans;
	}
}